src/PostgREST/DbRequestBuilder.hs

```diff
@@ -35,7 +35,7 @@ import           Protolude                hiding (from, dropWhile, drop)
 import           Text.Regex.TDFA         ((=~))
 import           Unsafe                  (unsafeHead)
 
-readRequest :: Maybe Integer -> [Relation] -> [(Text, Text)] -> ApiRequest -> Either Response ReadRequest
+readRequest :: Maybe Integer -> [Relation] -> [(Text, RetType)] -> ApiRequest -> Either Response ReadRequest
 readRequest maxRows allRels allProcs apiRequest  =
```

Comment:
The `allProcs` argument name seems kind of inaccurate, it's more like `procRetTypes`. Also maybe it's inefficient to do a list lookup each time. What if this argument was prepared ahead of time as a `Map Text RetType` rather than `[(Text, RetType)]`?

I realize the code was like this before your changes, but thought it would be a nice addition to the refactoring. if changing types to Map turns out to be pretty difficult you can skip that. :)